setup.sh: improve compatibility

rrthomas opened the pull request:
Remove --user flag. This is the default on some systems (Debian
derivatives), and is disabled on others (brew). Running without as the
default gives probably the best chance of annoying the fewest people.

rrthomas committed:
setup.sh: improve compatibility

Remove --user flag. This is the default on some systems (Debian
derivatives), and is disabled on others (brew). Running without as the
default gives probably the best chance of annoying the fewest people.

rrthomas committed:
README.md: simplify wording; rewrap for convenient viewing with “less”

rrthomas committed:
Scripts: various improvements

Make them directly runnable on systems that don’t use file extensions (add
hash-bang lines).

Add command-line usage, turning comments into easily-accessible
documentation.

Get arguments from command line, not environment variables.

Order imports into two blocks: built-in Python modules and the rest.

Print output from create-test-accounts.py as JSON to make it easier to
re-use programmatically. Print keys as strings, so the warning about
removing b'…' is redundant.

digitalronin commented:
Please update the README wrt. creating assets to match the changes you want to make to the script.

rrthomas committed:
Improve documentation, simplify code slightly

Add documentation for updated create-test-asset.py to README.md, explaining
the use of the command-line arguments.

“Refactor” README.md and run-server.sh to assume that either `pipenv run` or
`pipenv shell` is being used, and explain this under “Getting started”.

rrthomas commented:
Done, and I've said a bit more about `pipenv`, and assumed that you're going to either run `pipenv shell` or append `pipenv run` to everything yourself, thereby removing the need to have that in `run-server.sh`.
